java/maven/src/org/netbeans/modules/maven/api/execute/RunUtils.java

```diff
@@ -218,4 +219,10 @@ public static boolean hasTestCompileOnSaveEnabled(RunConfig config) {
         return false;
     }
 
+    private static void invokeLaterWithUI(Runnable runnable) {
+        if (GraphicsEnvironment.isHeadless()) {
+            return;
+        }
```

Comment:
This looks a bit strange - is this intended, that this invocation silently drops the execution of the runnable? I would be a bit surprised if trying to debug.

Reply:
Yes, the method `invokeLaterWithUI` is used to change cursor on main `JFrame` of the application. That yields an exception in headless mode. Not mangling the cursor shape - e.g. skipping the runnable - is exactly what is needed in headless mode.

Reply:
Understood - when I first read the method name, I assumed, that is intended to invoke the Runnable on the EDT. That made me wonder why `SwingUtilities#invokeLater` was not used, which led to me reading the method.